### ISSUE TITLE
bug: Top navigation - accessibility errors (ARC Toolkit extension)

### ISSUE THREAD

**Issue by vv-account** [cds-snc/design-gc-conception#2149]
### Prerequisites | Prérequis

- [x] I have read the [Contributing Guidelines](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md). | J’ai lu les [Lignes directrices sur les contributions](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md).
- [x] I agree to follow the [Code of Conduct](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md). | J’accepte de me conformer au [Code de conduite](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/cds-snc/gcds-components/issues) that already report this problem, without success. | J’ai vérifié parmi [les tickets (issues) existants](https://github.com/cds-snc/gcds-components/issues) que ce sujet n’avait pas déjà été soumis.

### GC Design System Components Package and Version | Paquet et version des composants de Système de design GC

gcds-components-vue@0.43.1

### Current Behavior | Comportement observé

ARC Toolkit (accessibility extension - version: 5.7.10) scan returns errors when scanning a page with Top navigation.

Our application uses the GCDS top navigation component. When we do an accessibility scan using ARC Toolkit, the report returns the same errors that are reported when scanning the GCDS top navigation example page: 

https://cds-snc.github.io/gcds-components/iframe.html?viewMode=docs&demo=true&singleStory=true&id=components-top-navigation--events-properties&lang=en

The error returned are as follows:

ARIA widget missing required context role
Description: The listitem role is missing the {{requiredContextRole}} required context role
WCAG 2.1 Success Criterion:A 4.1.2 Name, Role, Value : https://www.w3.org/TR/WCAG21/#name-role-value

List item has no list parent
Description: The `<li>` list item elements lack an appropriate `<ol>` (ordered list) or `<ul>` (unordered list) parent element.
WCAG 2.1 Success Criterion:A 4.1.1 Parsing : https://www.w3.org/TR/WCAG21/#parsing

No link text
Description: The link has no text.
WCAG 2.1 Success Criterion:A 2.4.4 Link Purpose (In Context) : https://www.w3.org/TR/WCAG21/#link-purpose-in-context

<img width="798" height="601" alt="Image" src="https://github.com/user-attachments/assets/c9a8cf26-4850-41e2-8bee-4e4082cb9d27" />

### Expected Behavior | Comportement attendu

No accessibility errors reported. 

### System Info | Information sur le système

```shell
The reported issue is reproducible on the GCDS top navigation page :
https://cds-snc.github.io/gcds-components/iframe.html?viewMode=docs&demo=true&singleStory=true&id=components-top-navigation--events-properties&lang=en

using the ARC Toolkit accessibility extension (Version: 5.7.10): 

https://chromewebstore.google.com/detail/arc-toolkit/chdkkkccnlfncngelccgbgfmjebmkmce?hl=en
```

### Steps to Reproduce | Étapes pour reproduire le bogue

1) Install ARC Toolkit accessibility extension in Google Chrome
https://chromewebstore.google.com/detail/arc-toolkit/chdkkkccnlfncngelccgbgfmjebmkmce?hl=en

2) In Chrome, navigate to https://cds-snc.github.io/gcds-components/iframe.html?viewMode=docs&demo=true&singleStory=true&id=components-top-navigation--events-properties&lang=en

3) Open dev tools in Chrome and click on the ARC Toolkit tab. (don't scan yet)

4) Click on the Elements tab, locate and click on "<gcds-top-nav label="Top navigation" alignment="right" class="hydrated">" which will make it the selected node

5) Click on the ARC Toolkit tab (in dev tools) and check the box for "Run on selected node only"

6) Click on Run tests

7) Accessibility report should be generated.

### Code Reproduction URL | URL de reproduction du code

https://cds-snc.github.io/gcds-components/iframe.html?viewMode=docs&demo=true&singleStory=true&id=components-top-navigation--events-properties&lang=en

### Additional Information | Informations supplémentaires

I suspect that either gcds-top-nav should have the attribute  role="list" added or "gcds-nav-link" shouldn't include the role="listitem" attribute.  I tried to use various role attributes and even thought added role="list" in our resolves some of the issues, it introduces new ones.
